lib/chai/utils/getPathInfo.js

```diff
@@ -61,14 +62,15 @@ module.exports = function getPathInfo(path, obj) {
  */
 
 function parsePath (path) {
-  var str = path.replace(/\[/g, '.[')
-    , parts = str.match(/(\\\.|[^.]+?)+/g);
-  return parts.map(function (value) {
-    var re = /\[(\d+)\]$/
-      , mArr = re.exec(value);
-    if (mArr) return { i: parseFloat(mArr[1]) };
-    else return { p: value };
-  });
+  /* assumed /^(?:(?:\\[.\[\]]|[^.\[\]])+|\[\d+\])(?:\.(?:\\[.\[\]]|[^.\[\]])+|\[\d+\])$/ */
+  var re = /(?:\\[.\[\]]|[^.\[\]])+|\[(\d+)\]/g
+    , parsed = []
+    , value;
+  while ((value = re.exec(path)) !== null) {
```

Comment:
I need to use `while` instead of `Array#map` to avoid matching /\[(\d+)\]/ twice.